cg/meta/demultiplex/validation.py

```diff
@@ -6,24 +6,25 @@
 from cg.exc import FlowCellError
 
 
-def validate_sample_fastq_file(sample_fastq: Path) -> None:
+def is_valid_sample_fastq_file(sample_fastq: Path, sample_id: str) -> bool:
     """
     Validate that the sample fastq file name is formatted as expected.
 
-    Since many different naming conventions are used, these are the only assumptions which can be made:
+    Assumptions:
     1. The sample fastq file name ends with .fastq.gz
     2. The sample fastq file name contains the lane number formatted as _L<lane_number>
-    3. The sample fastq file is located in a directory containing the sample id formatted as Sample_<sample_id>
+    3. The sample id is present in the parent directory name or in the file name.
     """
+    sample_id_in_directory = is_sample_id_in_directory_name(
+        directory=sample_fastq.parent, sample_id=sample_id
+    )
+    sample_id_in_file_name = is_sample_id_in_file_name(sample_fastq, sample_id)
 
-    if not is_file_path_compressed_fastq(sample_fastq):
-        raise ValueError(f"Sample fastq must end with {FileExtensions.FASTQ}{FileExtensions.GZIP}.")
-
-    if not is_lane_in_fastq_file_name(sample_fastq):
-        raise ValueError("Sample fastq must contain lane number formatted as '_L<lane_number>'.")
-
-    if not is_sample_id_in_directory_name(sample_fastq.parent):
-        raise ValueError("Parent directory name of sample fastq must contain 'Sample_<sample_id>'.")
+    return (
+        is_file_path_compressed_fastq(sample_fastq)
```

Comment:
```suggestion
        is_file_path_compressed_fastq(fastq_file_path)
```

Reply:
I think that is too verbose and it is clear by context that it is a path

Reply:
Ah, yeah the argument in the function definition for `is_file_path_compressed_fastq` is called `file_path`. But the function is called with a variable called `sample_fastq`. I don't want to call that variable `file_path` since the function specifically validates a sample fastq.

Reply:
Here are the signatures of the functions:

```python
is_valid_sample_fastq_file(sample_fastq: Path, sample_internal_id: str) -> bool
is_file_path_compressed_fastq(file_path: Path) -> bool
```